app/javascript/spec/ansible-repository-form/helpers.spec.js

```diff
@@ -0,0 +1,78 @@
+import { urlValidation } from '../../components/ansible-repository-form/helpers';
+
+const TESTCASES = [ // Example Cases
+  [{ scm_url: 'git@github.com:IonicaBizau/node-git-url-parse.git' }, true],
+  [{ scm_url: 'https://github.com/IonicaBizau/node-git-url-parse.git' }, true],
+  [{ scm_url: 'https://github.com/IonicaBizau/git-url-parse/blob/master/test/index.js' }, true],
+  [{ scm_url: '' }, false],
+  [{}, false], // url not set
+
+  // url scheme
+  [{ scm_url: '_' }, true],
```

Comment:
I'm surprised _any_ of here and below are actually true.  Why is that?  I'm curious if we even need these tests below or what they are meant to show.

Reply:
So most of these are passing as true because the parser identifies them as protocol `file`. Some of these cases pass because they're identified as protocol `ssh` with a non-empty path and that is treated as valid. Right now the parser is only checking to make sure that the given url has the appropriate protocol (i.e. http, https, file, ssh) or in the case of ssh, protocol and non-empty path).

Reply:
interesting...I'm wondering if we should change the rules to be http, https, file, ssh _and_ a path.  I can't imagine a case where a non-path would ever work.

Reply:
Ideally, I think we should make all of these from here down be false, because none of them are valid from our code's point of view.

Reply:
We can easily have path be a parameter for all the protocols (i do with [with ssh here](https://github.com/ManageIQ/manageiq-ui-classic/pull/8122/files#diff-3065271e4c9fe5a07dd211a26d147697afd8ac004bc515da1d82120368d811cfR11) we can just have it be like that for all) but then sites like `https://www.google.com/` would fail the url validation because it doesn't have a path, which is why i didn't configure it like that in the first place

Reply:
Pushed up changes to make path required.